server/asset/eth/rpcclient.go

```diff
@@ -27,16 +29,22 @@ type rpcclient struct {
 	ec *ethclient.Client
 	// c is a direct client for raw calls.
 	c *rpc.Client
+	// es is a wrapper for contract calls.
+	es *swap.ETHSwap
 }
 
 // connect connects to an ipc socket. It then wraps ethclient's client and
 // bundles commands in a form we can easil use.
-func (c *rpcclient) connect(ctx context.Context, IPC string) error {
+func (c *rpcclient) connect(ctx context.Context, IPC string, contractAddr *common.Address) error {
 	client, err := rpc.DialIPC(ctx, IPC)
 	if err != nil {
 		return fmt.Errorf("unable to dial rpc: %v", err)
 	}
 	c.ec = ethclient.NewClient(client)
+	c.es, err = swap.NewETHSwap(*contractAddr, c.ec)
+	if err != nil {
+		return fmt.Errorf("unable to find swap contract: %v", err)
+	}
```

Comment:
Feels a bit awkward and confusing that `contractAddr` is an input arg to `connect`.  Can `(*rpcclient).swap` have the contract as an input instead?  I think we'd need it as an arg to `swap` eventually when multiple contract addresses are possible, as would be needed for even contract upgrades.

Maybe I'm just fussing because there's no real `rpcclient` constructor, presently just `(*Backend) Connect` manually making the struct, and instead `connect` is being dual purposed.  These are all internals so I guess it's fine, but an API I would get would have a constructor that takes a slice/map of contracts to expect/support, then `connect` would setup an `ETHSwap` for each one using the provided IPC client, then `swap` would pick the one for the provided contract address.